### ISSUE TITLE
Crossref API returning an array under `Work.reference` field, and it’s unclear what `reference` means in this context now

### ISSUE THREAD

**Issue by strogonoff** [ietf-ribose/bibxml-service#208]
Per Crossref API specification, on a Work, `reference` is supposed to contain a single object (see Work under https://api.crossref.org/swagger-ui/index.html). Based on that, I assumed it to represent the work in question as well.

However, for `10.3390/fi11030055`, we’re getting an array of items, and judging by the content it looks to be a list of other works that reference the retrieved work.

This breaks the code, but more importantly if `Work.reference` means “references to this work from other works” then it shouldn’t be used at all.

Related to email report by @rjsparks, cc @kesara @ronaldtse

**Comment by strogonoff** [comment]
After removal of fallback on `Work.reference`, this now returns a successful response: https://dev.bibxml.org/public/rfc/bibxml7/reference.DOI.10.3390/fi11030055.xml

I’m not closing the issue yet just in case some light could be shed on the cause.

**Comment by rjsparks** [comment]
Have you looked at how doilit (https://github.com/cabo/kramdown-rfc/blob/f72472a62375c38f0f8a5e6c31068c2ee9002a98/bin/doilit) handles this?

**Comment by strogonoff** [comment]
@rjsparks

Thanks for the pointer.

* I suspect it doesn’t shed light on this particular bug, since it concerns DOI data used by BibXML service but not by doilit.

  This service converts received DOI item to a Relaton structure, and Relaton supports ISBN/ISSN identifiers among other things. Which is where I ran into issues where I did selective checks and sometimes Crossref’s `Work` structure wouldn’t have an ISBN directly, but it could be found on its `reference` property for some reason. I don’t know exactly why I decided using `Work.reference.isbn` was acceptable, but I think it was because `Work.reference` is declared to be a single object so I just never expected it to be anything else than a further Work metadata in some shape or form. In case of “incoming” references it’d naturally have to be an array. (And now it turns out for some items we _are_ getting an array, contrary to API spec.)

* However, looking at doilit, I see use of other DOI fields that doilit uses but BibXML service doesn’t. Namely, [`volume`, `journal-issue` and `page`](https://github.com/cabo/kramdown-rfc/blob/f72472a62375c38f0f8a5e6c31068c2ee9002a98/bin/doilit#L88-L97). I’ll file this.

**Comment by ronaldtse** [comment]
@strogonoff regarding https://dev.bibxml.org/public/rfc/bibxml7/reference.DOI.10.3390/fi11030055.xml , we also have to normalise the `<jats:p>` element to the BibXML format. Right now there is already `<t>` for paragraphs but need to remove the `<jats:p>`.

```xml
<abstract>
  <t><jats:p>The fourth industrial revolution heralds a paradigm shift in how people...</jats:p></t>
</abstract>
```

**Comment by strogonoff** [comment]
@ronaldtse we can address this particular issue, but just in case—we do not provide guarantees that we extract paragraphs correctly. We can sort of achieve that with Relaton data generated from authoritative sources, where @andrew2net’s code specifies correct format, but Crossref’s DOI API declares abstract as simply “string” so in addition to HTML and JATS XML we could feasibly encounter Markdown, LaTeX, or even a mix of formats. We can attempt a naive detection of format (https://github.com/ietf-ribose/bibxml-service/issues/210), but we can’t test for everything.

**Comment by strogonoff** [comment]
@ronaldtse Half the job is detecting MIME type for abstracts returned by Crossref API response, that part is in BibXML service’s. Other half of the job is splitting abstracts into paragraphs based on MIME type, that part is relaton-py’s.

I must correct myself in that we _can_ represent markup for definition, ordered and unordered lists in addition to paragraphs in xml2rfc (https://xml2rfc.tools.ietf.org/xml2rfc-doc.html#section-2.1-3.2). But, of course, we can’t represent anything beyond that.

**Comment by strogonoff** [comment]
Anyway, let’s not discuss other stuff here, the question is about `reference` and it’s not been addressed yet

**Comment by rjsparks** [comment]
I'm not sure what the question is?
What crossref is returning are documents that the document in question references (if the people that posted the document to crossef included that metadata.
You can see this directly by looking at the references section of
https://mdpi-res.com/d_attachment/futureinternet/futureinternet-11-00055/article_deploy/futureinternet-11-00055.pdf?version=1551185795
and what's returned by
https://api.crossref.org/v1/works/10.3390/fi11030055

I'm not sure why _our_ service's bib information for a document would be trying to include what the document referenced?

**Comment by strogonoff** [comment]
@rjsparks Thanks for looking into this, that was indeed the question.… In my mind it could be incoming references or the same work formatted as a “reference”. For some reason I decided it’s the latter, because API spec limited it to one object.

@ronaldtse It seems like it’s Crossref’s API spec problem after all. Either the API spec I consulted is outdated, or they shouldn’t specify it as a single object. (Ideally their API spec should also describe what different fields and objects mean, maybe they do it elsewhere.)

No action is needed, since references are already no longer mistakenly used the way they were.
